sync-core/src/main/java/com/cloudant/sync/query/IndexManager.java

```diff
@@ -226,35 +244,46 @@ public String ensureIndexed(List<Object> fieldNames) {
      *  @return name of created index
      */
     public String ensureIndexed(List<Object> fieldNames, String indexName) {
-        return ensureIndexed(fieldNames, indexName, "json");
+        return IndexCreator.ensureIndexed(Index.setUp(fieldNames, indexName),
```

Comment:
`setUp` creates an Index object rather than setting up the index itself. Is this class method -> instance of object pattern typically given the name `setUp`? E.g., in objc there's the convention of `[Index indexNamed:indexName forFields:fieldNames]` -- is there a convention like `Index.indexFor(fieldNames, indexName)`?